File: app/src/main/java/br/com/nicetuber/CustomApplication.java
package br.com.nicetuber;

import android.app.Application;

import br.com.nicetuber.modules.AndroidModule;
import br.com.nicetuber.modules.ApiModule;
import br.com.nicetuber.modules.NetModule;
import br.com.nicetuber.modules.RetrofitModule;

/**
 * Created by thiagozg on 15/10/2017.
 */

public class CustomApplication extends Application {

    private AppComponent appComponent;

    @Override
    public void onCreate() {
        super.onCreate();

        appComponent = DaggerAppComponent.builder()
                        .androidModule(new AndroidModule(this))
                        .netModule(new NetModule())
                        .retrofitModule(new RetrofitModule())
                        .apiModule(new ApiModule())
                        .build();
    }

    public AppComponent getAppComponent() {
        return appComponent;
    }

}

File: app/src/main/java/br/com/nicetuber/CustomApplication.kt
package br.com.nicetuber

import android.app.Application
import br.com.nicetuber.modules.AndroidModule
import br.com.nicetuber.modules.ApiModule
import br.com.nicetuber.modules.NetModule
import br.com.nicetuber.modules.RetrofitModule

/**
 * Created by thiagozg on 15/10/2017.
 */

class CustomApplication : Application() {


    lateinit var appComponent: AppComponent

    override fun onCreate() {
        super.onCreate()

        appComponent = DaggerAppComponent.builder()
                .androidModule(AndroidModule(this))
                .netModule(NetModule())
                .retrofitModule(RetrofitModule())
                .apiModule(ApiModule())
                .build()
    }

}
